packages/hydrogen/src/vite/plugin.ts

```diff
@@ -68,6 +68,11 @@ export function hydrogen(pluginOptions: HydrogenPluginOptions = {}): Plugin[] {
               ],
             },
           },
+          // Vite performs an initial reload after optimizing these dependencies.
+          // Do it early to avoid the initial reload:
+          optimizeDeps: {
+            include: ['@shopify/hydrogen'],
+          },
```

Comment:
Vite decides to prebundle Hydrogen (I'm not really sure why) when it's used outside of our monorepo. You can see this in the demo-store:

<img width="941" alt="image" src="https://github.com/Shopify/hydrogen/assets/1634092/d0c643ff-d551-4e5b-83af-b228e3e9f5be">

This just lets Vite find the dependency and prebundle it earlier to avoid the initial refresh.